luarules/gadgets/unit_maverick_exp.lua

```diff
@@ -20,39 +20,42 @@ local GetUnitExperience = Spring.GetUnitExperience
 local SetUnitMaxRange = Spring.SetUnitMaxRange
 local GetUnitDefID = Spring.GetUnitDefID
 
-local isMaverick = {}
-for udid, ud in pairs(UnitDefs) do
-	if string.find(ud.name, 'armmav') then
-		isMaverick[udid] = true
-	end
-end
-local maverickOriginalRange = WeaponDefNames["armmav_armmav_weapon"].range
-
 local updateList = {}
 
 function gadget:GameFrame(n)
 	for unitID in pairs(updateList) do
 		local curExp = GetUnitExperience(unitID)
         if curExp then
-            local limExp = ((3*curExp)/(1+3*curExp))*0.6
-            local newRange = maverickOriginalRange * ( 1 + limExp )
+			local unitDef = UnitDefs[GetUnitDefID(unitID)]
+
+			local rangeXPScale = unitDef.customParams.rangexpscale
+            local limExp = ((3*curExp)/(1+3*curExp))*rangeXPScale
+
+            local newRange = WeaponDefs[unitDef.weapons[1].weaponDef].range  * ( 1 + limExp )
+
             SetUnitWeaponState(unitID, 1, "range", newRange)
             SetUnitMaxRange(unitID,newRange)
         end
     end
 	updateList = {}
 end
 
-function GG.requestMaverickExpUpdate(unitID)
-	if not isMaverick[GetUnitDefID(unitID)] then
+function GG.requestMaverickExpUpdate(unitID)	
+	if (not(string.find(UnitDefs[attackerDefID].name, "armmav"))) then
 		return
 	end
+
 	--schedule an update of range next frame when exp has been updated
     updateList[unitID] = true
 end
 
 function gadget:UnitDamaged(unitID, unitDefID, unitTeam, damage, paralyzer, weaponDefID, projectileID, attackerID, attackerDefID, attackerTeam)
-	if isMaverick[attackerDefID] then
+	--Check if it doesn't exist anymore
+	if(UnitDefs[attackerDefID] == nil) then
+		return
+	end
+
+	if(string.find(UnitDefs[attackerDefID].name, "armmav")) then
```

Comment:
Same as above, check for customparam rather than string match

Reply:
Fixed.